includes/helper/class-wp-job-manager-helper.php

```diff
@@ -545,39 +545,41 @@ function ( $plugin ) use ( $search_term ) {
 	/**
 	 * Return if we should show or not the bulk activation form.
 	 *
-	 * @param array $licenced_plugins The list of licensed plugins to handle.
+	 * @param array $licenced_plugins The list of licenced plugins to handle.
 	 *
 	 * @return bool If we should show the bulk activation form or not.
 	 */
 	private function show_bulk_activation_form( $licenced_plugins ) {
 		foreach ( array_keys( $licenced_plugins ) as $product_slug ) {
 			$licence = self::get_plugin_licence( $product_slug );
-			if ( empty( $licence['licence_key'] ) && apply_filters( 'wpjm_display_license_form_for_addon', true, $product_slug ) ) {
+			if ( empty( $licence['licence_key'] ) && apply_filters( 'wpjm_display_licence_form_for_addon', true, $product_slug ) ) {
 				return true;
 			}
 		}
 		return false;
 	}
 
 	/**
-	 * Outputs unset license key notices.
+	 * Outputs unset licence key notices.
 	 */
-	public function licence_error_notices() {
-		$screen = get_current_screen();
-		if (
-			null === $screen ||
-			in_array( $screen->id, [ 'job_listing_page_job-manager-addons' ], true ) ||
-			! current_user_can( 'update_plugins' )
-		) {
-			return;
-		}
+	public function maybe_add_licence_error_notices() {
 		foreach ( $this->get_installed_plugins() as $product_slug => $plugin_data ) {
 			$licence = $this->get_plugin_licence( $product_slug );
 			if ( ! WP_Job_Manager_Helper_Options::get( $product_slug, 'hide_key_notice' ) ) {
 				if ( empty( $licence['licence_key'] ) ) {
-					include 'views/html-licence-key-notice.php';
+					add_filter(
+						'wpjm_admin_notices',
+						function ( $notices ) use ( $product_slug, $plugin_data ) {
+							return $this->add_missing_licence_notice( $notices, $product_slug, $plugin_data );
+						}
+					);
 				} elseif ( ! empty( $licence['errors'] ) ) {
-					include 'views/html-licence-key-error.php';
+					add_filter(
+						'wpjm_admin_notices',
+						function ( $notices ) use ( $product_slug, $plugin_data ) {
```

Comment:
I tried to avoid the anonymous functions here, but I'm not aware of another way to pass `$product_slug` and `$plugin_data` to the methods called.